src/db/tileDatabase.ts

```diff
@@ -0,0 +1,28 @@
+import * as sqlite3 from "sqlite3";
+import { TilesRow } from "../types/db";
+
+export class TileDatabase {
+    db: sqlite3.Database;
+
+    constructor(dbFileName: string) {
+        this.db = new sqlite3.Database(dbFileName, sqlite3.OPEN_READONLY);
+    }
+
+    getTileData(z: number, x: number, y: number): object {
+        return new Promise((resolve, reject) => {
+            this.db.get(
+                `SELECT tile_data FROM tiles WHERE zoom_level = ${z} AND tile_column = ${x} AND tile_row = ${y}`,
+                (err, row) => {
+                    if (err) {
+                        reject(err);
+                    }
+                    if (row == undefined) {
+                        resolve(null);
+                    } else {
+                        resolve((row as TilesRow).tile_data);
```

Comment:
I'm struggling to reconcile the fact that the SELECT query looks for a 'tile_data' using both row and column, but the return object seems to be at the row level - I was expecting to see a single tile.
Is this perhaps using 'row' in the sense of 'row in a table' rather than 'row on a grid of tiles'?

Reply:
Correct - see above comment.